src/hotspot/share/gc/shared/referenceProcessor.hpp

```diff
@@ -596,42 +585,15 @@ class ReferenceProcessorAtomicMutator: StackObj {
   }
 };
 
-// This class is an interface used to implement task execution for the
-// reference processing.
-class AbstractRefProcTaskExecutor {
-public:
-
-  // Abstract tasks to execute.
-  class ProcessTask;
-
-  // Executes a task using worker threads.
-  virtual void execute(ProcessTask& task, uint ergo_workers) = 0;
-};
-
-// Abstract reference processing task to execute.
-class AbstractRefProcTaskExecutor::ProcessTask {
-protected:
-  ReferenceProcessor&           _ref_processor;
-  // Indicates whether the phase could generate work that should be balanced across
-  // threads after execution.
-  bool                          _marks_oops_alive;
-  ReferenceProcessorPhaseTimes* _phase_times;
-
-  ProcessTask(ReferenceProcessor& ref_processor,
-              bool marks_oops_alive,
-              ReferenceProcessorPhaseTimes* phase_times)
-    : _ref_processor(ref_processor),
-      _marks_oops_alive(marks_oops_alive),
-      _phase_times(phase_times)
-  { }
+enum class ThreadModel {Multi, Single};
 
+class AbstractClosureContext {
```

Comment:
Could the `AbstractClosureContext` be renamed to something like `AbstractRefProcClosureContext` (remove the `Abstract` if too long - it does not really carry a lot of information). The current name is too generic and does not indicate in any way where it belongs to.

Reply:
I will rename to AbstractRefProcClosureContext.